src/main/java/searchengine/booleanretrieval/DataSelection.java

```diff
@@ -82,35 +82,13 @@ private String buildQuery(String input, String[] sources) { //TODO: Check if UI
         query.append(String.format("wordname='%s' ", terms[i]));
       }
     }
-    query.append(") ");
-    for (String source : sources) { //Search in selected sources (databases)
-      query.append(String.format("AND source='%s'", source)); //TODO: replace 'source' with the actual attribute
-    }
-    return query.toString();
-  }
-*/
-
-  /** SubFunction in retrieveDocuments. Generates an SQL query that searches for the terms from the input string in the wordcount database.
-   * The input string should be split with space between terms.
-   * @param input: the search query in the format "term1 term2 ... termN" (space separated)
-   * @return the SQL query
-   */
-  private String buildQuery(String input) {
-    StringBuilder query = new StringBuilder();
-    query.append("SELECT DISTINCT wordname, amount, articletitle, filepath "); //Distinct: select unique value
-    query.append("FROM wordratios ");
-    query.append("WHERE articletitle ");
-    query.append("IN ( SELECT articletitle ");
-    query.append("FROM wordratios WHERE ");
-    String[] terms = input.split("\\s+"); //Split on whitespace
-    for (int i = 0; i < terms.length; i++) { //Append terms
-      if (i < terms.length - 1) {
-        query.append(String.format("wordname='%s' OR ", terms[i]));
-      } else {
-        query.append(String.format("wordname='%s' ", terms[i]));
-      }
+    query.append(")");
+    if(sources.length > 0) {
+      query.append(" AND (sourcename = '");
+      query.append(String.join("' or sourcename = '", sources));
```

Comment:
Capitalize the or's here for readability?

Reply:
```suggestion
      query.append(String.join("' OR sourcename = '", sources));
```